src/overlays/actors/ovl_En_Guruguru/z_en_guruguru.c

```diff
@@ -21,32 +45,358 @@ const ActorInit En_Guruguru_InitVars = {
+}
+
+void EnGuruguru_DoNothing(EnGuruguru* this, GlobalContext* globalCtx) {
+}
+
+void func_80BC6E10(EnGuruguru* this) {
+    EnGuruguru_ChangeAnimation(this, 0);
+    this->textIDIndex = 0;
+    this->unk270 = 0;
+    if (this->actor.params == 0) {
+        if (gSaveContext.weekEventReg[38] & 0x10) {
+            this->textIDIndex = 1;
+        }
+    } else if (gSaveContext.weekEventReg[38] & 0x40) {
+        this->textIDIndex = 2;
+    } else {
+        this->textIDIndex = 3;
+        this->unk270 = 1;
+    }
+    this->headZRotTarget = 0;
+    this->unk268 = 1;
+    this->actor.textId = textIDs[this->textIDIndex];
+    if (((this->textIDIndex == 0) || (this->textIDIndex == 1)) && ((gSaveContext.weekEventReg[77] & 4))) {
+        if ((gSaveContext.weekEventReg[88] & 4) == 0) {
+            this->actor.textId = 0x295F;
+        } else {
+            this->actor.textId = 0x2960;
+        }
+    }
+    this->unk272 = 0;
+    this->actionFunc = func_80BC6F14;
+}
+
+void func_80BC6F14(EnGuruguru* this, GlobalContext* globalCtx) {
+    s16 yaw;
+    s16 yawTemp;
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if (this->unk270 != 0) {
+        Player* player = PLAYER;
+
+        this->textIDIndex = 3;
+        if (player->transformation == PLAYER_FORM_DEKU) {
+            this->textIDIndex = 13;
+            if (gSaveContext.weekEventReg[79] & 4) {
+                this->textIDIndex = 14;
+            }
+        }
+
+        this->actor.textId = textIDs[this->textIDIndex];
+    }
+
+    yawTemp = this->actor.yawTowardsPlayer - this->actor.world.rot.y;
+    yaw = ABS_ALT(yawTemp);
+
+    if (func_800B84D0(&this->actor, globalCtx)) {
+        func_80BC701C(this, globalCtx);
+    } else if (yaw < 0x2891) {
+        func_800B8614(&this->actor, globalCtx, 60.0f);
+    }
+}
+
+void func_80BC701C(EnGuruguru* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if ((this->unk268 != 0) &&
+        ((player->transformation == PLAYER_FORM_HUMAN) || (player->transformation == PLAYER_FORM_DEKU))) {
+        this->headZRotTarget = 5000;
+    }
+
+    this->unk266 = 1;
+    this->unk272 = 1;
+    this->actionFunc = func_80BC7068;
+}
+
+void func_80BC7068(EnGuruguru* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Guruguru_0x80BC6BF0/EnGuruguru_Destroy.asm")
+    if (this->unk268 != 0) {
+        SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    } else if (this->unusedTimer == 0) {
+        this->unusedTimer = 6;
+        if (func_80152498(&globalCtx->msgCtx) != 5) {
+            if (this->unk266 == 0) {
+                if (this->headZRotTarget != 0) {
+                    this->headZRotTarget = 0;
+                } else {
+                    this->headZRotTarget = 5000;
+                }
+            }
+        } else {
+            if ((player->transformation == PLAYER_FORM_HUMAN) || (player->transformation == PLAYER_FORM_DEKU)) {
```

Comment:
```suggestion
            if (player->transformation == PLAYER_FORM_HUMAN || player->transformation == PLAYER_FORM_DEKU) {
```

Reply:
I think we do normally leave brackets around this sort of thing.